emcc.py

```diff
@@ -1429,6 +1444,22 @@ def save_intermediate(name=None, suffix='js'):
       file_code = execute([shared.PYTHON, shared.FILE_PACKAGER, unsuffixed(target) + '.data'] + file_args, stdout=PIPE)[0]
       pre_js = file_code + pre_js
 
+    if shared.Settings.BINARYEN:
+      # add in the glue integration code as a pre-js, so it is optimized together with everything else
+      wasm_js_glue = open(os.path.join(shared.BINARYEN_ROOT, 'src', 'js', 'wasm.js-post.js')).read()
+      wasm_js_glue = wasm_js_glue.replace("Module['asmjsCodeFile']", '"' + os.path.basename(asm_target) + '"') # " or '? who knows :)
```

Comment:
Generally we haven't done this kind of template replacing before. This only replaces first occurrence and not if it referenced multiple times, and won't replace `Module.asmjsCodeFile`, and it might be surprising to reader that we are doing it like this. Could we do the usual `{{{ xxx }}}` type of expansion of a template, or prepend some variables that the later code then accesses?